database/Reservation.js

```diff
@@ -2,102 +2,174 @@ const mongoose = require('./connect.js');
-            nameMatch = true;
-            break;
+            overlaps.push(reservations[i]);
+            //  Does the name match?  If so, decline "you're alread here"
+            if (booking.name !== '' && booking.name === reservations[i].customer_name) {
+              nameMatch = true;
+              break;
+            }
           }
         }
-      }
-      //  it works!
-      var notification = '';
-      if (nameMatch) {
-        notification = "Sorry, you already have a booking at this time.  Please make another selection.";
-      } else if (overlaps.length >= AVAIL_TABLES) {
-        nofitication = "Sorry, there are no bookings available on this date and time.  Please make another selection.";
-      } else {
-        //  add reservation to the database
-        notification = 'Your reservation has been successfully booked!';
-        addReservation(booking);
-      }
+        //  it works!
+        var notification = '';
+        if (nameMatch) {
+          notification = 'Sorry, you already have a booking at this time.  Please make another selection.';
+        } else if (overlaps.length >= AVAIL_TABLES) {
+          nofitication = 'Sorry, there are no bookings available on this date and time.  Please make another selection.';
+        } else {
+          //  add reservation to the database
+          notification = 'Your reservation has been successfully booked!';
+          addReservation(booking);
+        }
 
-      resolve(notification)
-    });
+        resolve(notification)
+      });
   });
 };
 
+//Ron's crud update: here is were I'm going to insert a put/update request to update/change a reservation
+
 addReservation = (booking) => {
-  var res = new Reservation(
-    {
-      restaurant_id: booking.restId,
-      customer_name: booking.name,
-      reservation_time: booking.time,
-      guests: booking.guests
-    });
+  var res = new Reservation({
+    reservation_id: booking.reservId,
+    restaurant_id: booking.restId,
+    customer_name: booking.name,
+    reservation_time: booking.time,
+    guests: booking.guests
+  });
 
   // save model to database
   res.save(function (err, booking) {
     if (err) return console.error(err);
   });
 };
 
+//!! RON'S ADDITION FOR THE PUT REQUEST TO UPDATE THE RESERVATION
+//format is: findByIdAndUpdate(id, {update}, callback)
+updateReservation = (booking, callback) => {
+  //console.log("this is the booking:", booking);
+  console.log(booking._id);
+  let query = Reservation.findByIdAndUpdate(
+    booking._id,
+    { guests: booking.guests },
+    function(err, result) {
+      if (err) {
+        return err;
+      } else {
+        return result;
+      }
+    }
+  );
+  return query.exec();
+};
+
+//!! END OF THE CHANGE FOR THE PUT REQUEST
+
+//!! RON'S ADDITION FOR THE DELETE REQUEST TO UPDATE THE RESERVATION
+//format is: findByIdAndUpdate(id, {update}, callback)
+deleteReservation = (booking, callback) => {
+  //console.log("this is the booking:", booking);
```

Comment:
looks good, but you can prob delete callback as you're not using it?

also don't forget to export it